### ISSUE TITLE
False positive USO_UNSAFE_ACCESSIBLE_OBJECT_SYNCHRONIZATION for locks created in `clone()`.

### ISSUE THREAD

**Issue by attipaci** [spotbugs/spotbugs#4320]
**Describe the bug**

When using a private synchronization lock object in classes, SpotBugs reports an USO_UNSAFE_ACCESSIBLE_OBJECT_SYNCHRONIZATION when clones have their own private locks created in the `clone()` implementation.

**To Reproduce**

1. Run Spotbugs on the sample code provided below.
2. It produces the following false positive:

```
[ERROR] Medium: The lock object nom.tam.fits.Foo.lock used in method nom.tam.fits.Foo.bar() is exposed to untrusted code outside of class nom.tam.fits.Foo, through updating or returning its value from nom.tam.fits.Foo.clone()Lnom/tam/fits/Foo;. [nom.tam.fits.Foo] At Foo.java:[lines 52-55] USO_UNSAFE_ACCESSIBLE_OBJECT_SYNCHRONIZATION
``` 

```java
public class Foo implements Cloneable {

    private int property; // class property with synchronized access

    private Object lock = new Object(); // private synchronization lock

    @Override
    protected Foo clone() {
        try {
            Foo clone = (Foo) super.clone();
            clone.lock = new Object(); // cloned object to have its own independent synchronization lock.
            return clone;
        } catch (CloneNotSupportedException e) {
            return null;
        }
    }

    public void bar() {
        synchronized (lock) { // Here we get an USO_UNSAFE_ACCESSIBLE_OBJECT_SYNCHRONIZATION
            property = 0;
        }
    }
}
```

**Expected behavior**

No USO_UNSAFE_ACCESSIBLE_OBJECT_SYNCHRONIZATION error, since the lock is never exposed.

**Environment**
- Java Compiler: OpenJDK
- Java version: JDK 25.0.4.1
- SpotBugs Version: 4.10.4.0
- How do you use SpotBugs: Maven Plugin
    - Plugin version: SpotBugs Maven Plugin 4.10.4.0
    - Environment version for Plugin: 3.9.11

**Additional context**

Note, that clones should always have their own private locks, or else the original class may block operations on the clone and vice-versa. SpotBugs does not complain if `clone()` retains the reference to the original lock object, even though that is a potential synchronization issue, since the clone and the original can block each other.
